packit_service/worker/helpers/testing_farm.py

```diff
@@ -60,40 +61,94 @@ class CommentArguments:
-    identifier: str = None
-    labels: List[str] = None
-    pr_argument: str = None
-
     def __init__(self, command_prefix: str, comment: str):
+        self.parser = None
+        self.packit_command: str = None
+        self.identifier: str = None
+        self.labels: List[str] = None
+        self.pr_argument: str = None
+        self.envs: Dict[str, str] = None
+
         if comment is None:
             return
 
-        # Try to parse identifier argument from comment
+        # Init argparse
+        self.parser = self.init_arg_parse
+
+        # Remove the command prefix from the comment
         logger.debug(f"Parsing comment -> {comment}")
         logger.debug(f"Used command prefix -> {command_prefix}")
 
-        match = re.search(
-            r"^" + re.escape(command_prefix) + r"\s(?P<packit_command>\S+)", comment
-        )
+        # Match the command prefix and extract the rest of the comment
+        match = re.match(r"^" + re.escape(command_prefix) + r"\s+(.*)", comment)
         if match:
-            self.packit_command = match.group("packit_command")
-            logger.debug(f"Parsed packit_command: {self.packit_command}")
+            arguments_str = match.group(1)
+        else:
+            # If the command prefix is not found, nothing to parse
+            logger.debug("Command prefix not found in the comment.")
+            return
 
-        match = re.search(r"(--identifier|--id|-i)[\s=](?P<identifier>\S+)", comment)
-        if match:
-            self.identifier = match.group("identifier")
-            logger.debug(f"Parsed test argument -> identifier: {self.identifier}")
+        # Use shlex to split the arguments string into a list
+        args_list = shlex.split(arguments_str)
+        logger.debug(f"Arguments list after shlex splitting: {args_list}")
 
-        match = re.search(r"--labels[\s=](?P<labels>\S+)", comment)
-        if match:
-            self.labels = match.group("labels").split(",")
-            logger.debug(f"Parsed test argument -> labels: {self.labels}")
+        # Parse known arguments
+        try:
+            args, unknown_args = self.parser.parse_known_args(args_list)
+            logger.debug(f"Parsed known args: {args}")
+            logger.debug(f"Unknown args: {unknown_args}")
+        except argparse.ArgumentError as e:
+            logger.error(f"Argument parsing error: {e}")
+            return
 
-        match = re.search(r"(?P<pr_arg>[^/\s]+/[^#]+#\d+)", comment)
-        if match:
-            self.pr_argument = match.group("pr_arg")
-            logger.debug(f"Parsed test argument -> pr_argument: {self.pr_argument}")
+        self.parse_known_arguments(args)
+        self.parse_unknown_arguments(unknown_args)
+
+    @property
+    def init_arg_parse(self):
+        # Set up argparse
+        parser = argparse.ArgumentParser()
+        parser.add_argument("packit_command")
+        parser.add_argument("--identifier", "--id", "-i")
+        parser.add_argument("--labels")
```

Comment:
```suggestion
        parser.add_argument("--labels", type=lambda s: s.split(","))
```

This should make `args.labels` below to already be a list.